### PR TITLE
Add deprecated metric utility functions back to functional

### LINKED ISSUE #5061: Fix metrics backwards compatibility
Moved to pl.metrics.utils (some users have complained about this)

TODO: add function back to functional (with deprecation warning) that simply calls new one

### PR DESCRIPTION
**Pull request by teddykoker**
## What does this PR do?

Fixes #5061

### CODE REVIEW

File: pytorch_lightning/metrics/functional/deprecated.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from pytorch_lightning.metrics.utils import to_categorical as new_to_categorical
+from pytorch_lightning.metrics.utils import to_onehot as new_to_onehot
+
+import warnings
+
+
+# TODO: remove in 1.1.1
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/deprecated.py:7]: 1.1.1 is the next release so it won't be for any release... so remove in 1.2?

File: pytorch_lightning/metrics/functional/deprecated.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from pytorch_lightning.metrics.utils import to_categorical as new_to_categorical
+from pytorch_lightning.metrics.utils import to_onehot as new_to_onehot
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/deprecated.py:2]:
```suggestion
from pytorch_lightning.metrics.utils import to_categorical as __to_categorical
from pytorch_lightning.metrics.utils import to_onehot as __to_onehot
```

File: pytorch_lightning/metrics/functional/deprecated.py (hunk 1 above)
Comment (teddykoker) [reply pytorch_lightning/metrics/functional/deprecated.py:7]: Will do

**Review by Borda — APPROVED** [review]
lgtm

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/metrics/functional/__init__.py [hunk 3]
```diff
@@ -38,3 +38,4 @@
 from pytorch_lightning.metrics.functional.roc import roc
 from pytorch_lightning.metrics.functional.self_supervised import embedding_similarity
 from pytorch_lightning.metrics.functional.ssim import ssim
+from pytorch_lightning.metrics.utils import to_categorical, to_onehot
```
Comment (rohitgr7) [inline pytorch_lightning/metrics/functional/__init__.py:41]: maybe add a TODO to remove it after v xx.x?

**Comment by teddykoker** [comment]
wait who edited it? and removed the deprecated warnings?

**Pull request merged**